Test/E2ETests/E2ETests/Test452Base.cs

```
@@ -0,0 +1,1066 @@
+            PopulateIPAddresses();
+
+            bool allAppsHealthy = HealthCheckAndRemoveImageIfNeededAllApp();
+
+            if (!allAppsHealthy)
+            {
+                DockerUtils.ExecuteDockerComposeCommand("up -d --build", DockerComposeFileName);
+                Thread.Sleep(5000);
+                DockerUtils.PrintDockerProcessStats("Docker-Compose -build retry");
+
+                allAppsHealthy = HealthCheckAndRemoveImageIfNeededAllApp();
+            }            
+
+            Assert.IsTrue(allAppsHealthy, "All Apps are not unhealthy.");            
+
+            dataendpointClient = new DataEndpointClient(new Uri("http://" + Apps[IngestionName].ipAddress));
+
+            InitializeDatabase();
+
+            Thread.Sleep(5000);
+            Trace.WriteLine(".Completed ClassInitialize:" + DateTime.UtcNow.ToLongTimeString());
+        }
+
+        private static void GetDockerCompose()
+        {
+            HttpClient client = new HttpClient();
+            var stream = client.GetStreamAsync("https://github.com/docker/compose/releases/download/1.12.0/docker-compose-Windows-x86_64.exe").Result;
+            if (!File.Exists(".\\docker-compose.exe"))
+            {
+                FileStream fs = null;
+                try
+                {
+                    fs = new FileStream(".\\docker-compose.exe", FileMode.Create, FileAccess.Write, FileShare.None);
+                    stream.CopyToAsync(fs).Wait();
+                }
+                finally
+                {
+                    fs.Close();
+                }
+            }
+        }
+
+        private static void InitializeDatabase()
+        {
+            var ip = DockerUtils.FindIpDockerContainer("e2etests_sql-server_1");
+            LocalDbHelper localDbHelper = new LocalDbHelper(ip);
+            if (!localDbHelper.CheckDatabaseExists("dependencytest"))
+            {
+                localDbHelper.CreateDatabase("dependencytest", "c:\\dependencytest.mdf");
```

This is one of the problems I was trying to fix in my PR.

Please add a follow-up check to determine if the CreateDatabase failed.
Otherwise it will continue to execute all the SQL tests and we would rather this fail immediately.

```
if (!LocalDbHelper.CheckDatabaseExists("dependencytest")) 
{ 
throw new Exception($"Failed to create database: 'dependencytest'"); 
}
```

---

If any sql error occurs, exception is not handled and would cause test initialize to fail. No tests will be run at all.

---

if (!LocalDbHelper.CheckDatabaseExists("dependencytest")) 
{ 
throw new Exception($"Failed to create database: 'dependencytest'"); 
}

This check can be added to make it explciity why test failed. Will do

---

added